tests/foreman/ui/test_host.py

```diff
@@ -1701,3 +1704,194 @@ def test_positive_delete_libvirt(
+            'locations.resources.assigned': [module_loc.name],
+        })
+    gce_cr = entities.AbstractComputeResource().search(
+        query={'search': 'name={}'.format(cr_name)})[0]
+    entities.Image(
+        architecture=default_architecture, compute_resource=gce_cr, name='autogce_img',
+        operatingsystem=module_os, username=vm_user, uuid=class_gce_template
+    ).create()
+    return gce_cr
+
+
+@pytest.fixture
+def class_gce_hostgroup(
+        module_org,
+        module_loc,
+        default_partition_table,
+        default_architecture,
+        module_os,
+        module_environment,
+        module_proxy,
+        class_gce_domain,
+        class_gce_resource_with_image,
+        module_lce,
+        module_content_view,
+):
+    return entities.HostGroup(
+        architecture=default_architecture,
+        compute_resource=class_gce_resource_with_image,
+        domain=class_gce_domain,
+        lifecycle_environment=module_lce,
+        content_view=module_content_view,
+        location=[module_loc],
+        environment=module_environment,
+        puppet_proxy=module_proxy,
+        puppet_ca_proxy=module_proxy,
+        content_source=module_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+        ptable=default_partition_table,
+    ).create()
+
+
+@tier4
+@skip_if_not_set('gce')
+def test_positive_gce_provision_end_to_end(
+        session,
+        module_org,
+        module_loc,
+        module_os,
+        class_gce_domain,
+        class_gce_hostgroup,
+        gce_client
+):
+    """Provision Host on GCE compute resource
+
+    :id: 8d1877bb-fbc2-4969-a13e-e95e4df4f4cd
+
+    :expectedresults: Host is provisioned successfully
+
+    :CaseLevel: System
+    """
+    name = gen_string('alpha').lower()
+    hostname = u'{0}.{1}'.format(name, class_gce_domain.name)
+    gceapi_vmname = hostname.replace('.', '-')
+    root_pwd = gen_string('alpha', 15)
+    with Session('gce_tests') as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision GCE Host
+        try:
+            session.host.create({
+                'host.name': name,
+                'host.hostgroup': class_gce_hostgroup.name,
+                'provider_content.virtual_machine.machine_type': 'g1-small',
+                'provider_content.virtual_machine.external_ip': True,
+                'provider_content.virtual_machine.network': 'default',
+                'operating_system.operating_system': module_os.title,
+                'operating_system.image': 'autogce_img',
+                'operating_system.root_password': root_pwd
+            })
+            wait_for(
+                lambda: session.host.get_details(hostname)[
+                            'properties']['properties_table']['Build'] != 'Pending installation',
+                timeout=600,
+                delay=30,
```

Comment:
Can we use some API way to check the host Build status ? and once status(!Pending installation) get from API move further ...
as that would help to save the delay time (because on UI this check always takes time to go through entire navigation)

Reply:
1. It is good to check 'Installed' status from UI as its UI test.
2. GCE doesnt supports API endpoint for now.

Reply:
You are right `Installed` should check from UI and same you could check in assertion under [1] ..

My thought here is  under wait_for we are checking each time through UI only(it is too slower then API) ... so same can check from API (by search the host and check status ... no GCE API support needed for that) ..

[1 ]https://github.com/SatelliteQE/robottelo/pull/7264/files#diff-339c21393e2e46d258dccc49edb476dbR1885